queasars/minimum_eigensolvers/base/evolving_ansatz_minimum_eigensolver.py

```diff
@@ -0,0 +1,471 @@
+                    population=population, operator_context=operator_context
+                )
+                if (
+                    self.configuration.max_circuit_evaluations is not None
+                    and estimated_evaluations is not None
+                    and n_circuit_evaluations + estimated_evaluations >= self.configuration.max_circuit_evaluations
+                ):
+                    terminate = True
+
+                # Terminate if the maximum amount of generations have passed
+                if (
+                    self.configuration.max_generations is not None
+                    and n_generations >= self.configuration.max_generations
+                ):
+                    terminate = True
+
+                if terminate:
+                    break
+
+                population = operator.apply_operator(population=population, operator_context=operator_context)
+
+        if (
+            current_best_individual is None
+            or current_best_expectation_value is None
+            or len(population_evaluations) == 0
+        ):
+            raise Exception("The algorithm seems to have terminated without having evaluated any population! ")
+
+        sampler_result_best_individual: SamplerResult = self.configuration.sampler.run(
+            current_best_individual.get_quantum_circuit().measure_all(inplace=False)
+        ).result()
+
+        result = EvolvingAnsatzMinimumEigensolverResult()
+        result.eigenvalue = current_best_expectation_value
+        result.eigenstate = sampler_result_best_individual.quasi_dists[0]
+        result.optimal_circuit = current_best_individual.get_parameterized_quantum_circuit()
+        result.optimal_parameters = dict(
+            zip(result.optimal_circuit.parameters, current_best_individual.get_parameter_values())
+        )
+        result.circuit_evaluations = n_circuit_evaluations
+        result.generations = n_generations
+        result.population_evaluation_results = population_evaluations
+
+        if isinstance(aux_circuit_evaluators, list):
+            result.aux_operators_evaluated = [
+                evaluator.evaluate_circuits(
+                    [current_best_individual.get_parameterized_quantum_circuit()],
+                    [list(current_best_individual.get_parameter_values())],
+                )[0]
+                for evaluator in aux_circuit_evaluators
+            ]
+        if isinstance(aux_circuit_evaluators, dict):
+            result.aux_operators_evaluated = {
+                name: evaluator.evaluate_circuits(
+                    [current_best_individual.get_parameterized_quantum_circuit()],
+                    [list(current_best_individual.get_parameter_values())],
+                )[0]
+                for name, evaluator in aux_circuit_evaluators.items()
+            }
+
+        return result
+
+
+class EvolvingAnsatzMinimumEigensolverResult(MinimumEigensolverResult):
```

Comment:
I am not sure about circular dependencies, but is it possible to move this to its own file? (Maybe in a later MR if that's easier.)

Reply:
Moving the result class to a new file was doable without issue. Take a look, if this suits you better 👍

Reply:
Perfect, thanks!